### PR TITLE
[Wapuu] Make error system more robust

### PR DESCRIPTION
**Pull request by AllTerrainDeveloper**
## Proposed Changes

If somehow we end up having errors from server without no body, client is unable to recover from the error state. This address it

## Why are these changes being made?
In the rare case where we receive 500 error from the server without body response, we couldn't recover the Wapuu client from the error state. This patch addresses it.

## Testing Instructions
Have a quick chat with Wapuu and ask to contact with happiness engineers. It should work as expected.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/odie-client/src/components/message/sources.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const Sources = ( { message }: { message: Message } ) => {
 			];
 		}
 		return [];
-	}, [ message?.context?.sources ] );
+	}, [ message?.context?.sources, trackEvent ] );
```
Comment (robfelty) [inline packages/odie-client/src/components/message/sources.tsx:32]: Why is this necessary?

**Review by robfelty — APPROVED** [review]
Tested and works as described. 

Changes look good to me. I did have one question, but not blocking.

File: packages/odie-client/src/components/message/sources.tsx (hunk 1 above)
Comment (AllTerrainDeveloper) [reply packages/odie-client/src/components/message/sources.tsx:32]: That's a dependencies array, inside a `useMemo` method.

`useMemo` is meant to memoize calculations, so for this case, we process the data once and we store it as `const sources;` If sources changes, we update the const for the next render. if trackEvent function changes, we need to re "memoize" the calculations. 

This won't happen (we change that trackEvent function through the lifetime of the chat), but the correct thing is to always include all the dependencies in the array. I realized it was t throwing a warning so I fixed it.

image

Comment (AllTerrainDeveloper) [reply packages/odie-client/src/components/message/sources.tsx:32]: So, in the really unlikely chance of changing the `trackEvent` function through the chat, all the events will be sent through the new function :)

Comment (robfelty) [reply packages/odie-client/src/components/message/sources.tsx:32]: Thanks for the explanation! That is very helpful.
